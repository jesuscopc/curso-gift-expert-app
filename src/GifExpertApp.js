import React, { useState } from 'react'
import { GifGrid } from './components/GifGrid';
import { AddCategory } from './components/AddCategory';

const GifExpertApp = ({ defaultCategories = []}) => {

  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifEXpertApp</h2>
      <AddCategory
        setCategorie= { setCategories }
      />
      <hr/>
        {/* <button onClick={ handleAdd }>Agregar</button> */}
      <ol>
        { categories.map( categorie => (
            <GifGrid
              key={ categorie } 
              categorie={ categorie }
            />
          ))
        }
      </ol>
    </>
  )
}

export default GifExpertApp;