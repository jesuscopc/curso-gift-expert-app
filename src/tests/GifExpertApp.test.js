import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Sui Test GifExpertApp', () => {
  
  const categories = ['One punch', 'Yu Gi Oh']

  test('should be render correctly', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid'))
  });
  
  test('should be render all gifgrid categories', () => {
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
  
})
