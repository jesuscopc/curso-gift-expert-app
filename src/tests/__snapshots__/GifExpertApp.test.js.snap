// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui Test GifExpertApp should be render all gifgrid categories 1`] = `
<Fragment>
  <h2>
    GifEXpertApp
  </h2>
  <AddCategory
    setCategorie={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categorie="One punch"
      key="One punch"
    />
    <GifGrid
      categorie="Yu Gi Oh"
      key="Yu Gi Oh"
    />
  </ol>
</Fragment>
`;

exports[`Sui Test GifExpertApp should be render correctly 1`] = `
<Fragment>
  <h2>
    GifEXpertApp
  </h2>
  <AddCategory
    setCategorie={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
